### PR TITLE
TransitionData intellisense

### PR DESCRIPTION
**Pull request by nicolaserny**
Related to #72

### CODE REVIEW

File: extension/snippets/xcomponent.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
 		"prefix": "xctransition",
 		"description": "Create a XComponent transition",
 		"body": [
-			"<TransitionData Id=\"${1:Index}\" Name=\"${2:Transition name}\" FromKey=\"${3:State + Id}\" ToKey=\"${4:State + Id}\" Type=\"${5|X,TimeOut,Standard|}\" ExecutionDelay=\"${6:0}\" SetCustomTimeout=\"${7|false,true|}\" TriggeringEvent=\"${8:Document Triggering event}\" UserSpecificRule=\"${9|false,true|}\">",
+			"<TransitionData Id=\"${1}\" Name=\"${2:Transition name}\" FromKey=\"${3}\" ToKey=\"${4}\" Type=\"${5|X,TimeOut,Standard|}\" ExecutionDelay=\"${6:0}\" SetCustomTimeout=\"${7|false,true|}\" TriggeringEvent=\"${8:Document Triggering event}\" UserSpecificRule=\"${9|false,true|}\">",
```
Comment (maurelio1234) [inline extension/snippets/xcomponent.json]: I wouldn't put non mandadory parameters like ExecutionDelay, SetCustomTimeout, UserSpecificRule inthe snippet.
Why not creating a snippet for each kind of transition?

File: extension/src/completion/componentCompletionItemProvider.ts [hunk 2]
```diff
@@ -43,6 +43,22 @@ export class ComponentCompletionItemProvider implements vscode.CompletionItemPro
                 { value: "InternalMember" }
             ])
         },
+        { tag: "TransitionData", attribute: "Id", provider: new IdCompletionProvider("TransitionData") },
+        {
+            tag: "TransitionData", provider: new BasicCompletionProvider([
+                { value: "Id" },
```
Comment (maurelio1234) [inline extension/src/completion/componentCompletionItemProvider.ts]: is it possible to add the timeout related fields only if the transition is of type TimeOut?

Comment (nicolaserny) [reply extension/src/completion/componentCompletionItemProvider.ts]: I don't think so. The snippet mechanism is not as "clever" as we would like...
According to other extensions, the solution is to create different snippets:
in our case, xctransition and xctimeouttransition
What do you think about it ?

Comment (nicolaserny) [reply extension/src/completion/componentCompletionItemProvider.ts]: lol it's exactly your proposal in your other comment ! +1

File: extension/snippets/xcomponent.json (hunk 1 above)
Comment (nicolaserny) [reply extension/snippets/xcomponent.json]: Good idea !
In this PR I was focused on intellisense (the snippet was created in a preview PR)
But it makes sense. I will update my PR with snippets for each kind of transition

Comment (maurelio1234) [reply extension/snippets/xcomponent.json]: And fork transition.

File: extension/src/completion/componentCompletionItemProvider.ts (hunk 2 above)
Comment (maurelio1234) [reply extension/src/completion/componentCompletionItemProvider.ts]: But this code is for autocompletion, not snippet, isn't it?
That's why I proposed different snippets, and type specific autocompletion.

Comment (nicolaserny) [reply extension/src/completion/componentCompletionItemProvider.ts]: Ok I understand. Currently this autocomplete just provides completion based on all available xml attributes.
If we want to add logic in this part, we can do it I think (just a more challenging regex lol)
Perhaps we can create another ticket for that. I would like to provide all the basic intellisense as soon as possible

File: extension/snippets/xcomponent.json (hunk 1 above)
Comment (nicolaserny) [reply extension/snippets/xcomponent.json]: Yes of course

Comment (nicolaserny) [reply extension/snippets/xcomponent.json]: done

File: extension/src/completion/componentCompletionItemProvider.ts (hunk 2 above)
Comment (nicolaserny) [reply extension/src/completion/componentCompletionItemProvider.ts]: new ticket #96

**Pull request merged**